packages/backend/src/core/ForcedTransactionMonitor.ts

```diff
@@ -0,0 +1,97 @@
+import { Hash256, Timestamp } from '@explorer/types'
+import promiseRetry from 'promise-retry'
+
+import {
+  ForcedTransaction,
+  ForcedTransactionsRepository,
+} from '../peripherals/database/ForcedTransactionsRepository'
+import { EthereumClient } from '../peripherals/ethereum/EthereumClient'
+
+export class ForcedTransactionMonitor {
+  constructor(
+    private readonly ethereumClient: EthereumClient,
+    private readonly forcedTransactionRepository: ForcedTransactionsRepository
+  ) {}
+
+  async startWatching() {
+    const transactions = await this.forcedTransactionRepository.getPending()
+    transactions.forEach((tx) => this.watchTransaction(tx))
+  }
+
+  private async waitForTransaction(hash: Hash256) {
+    try {
+      return await promiseRetry(
+        (retry) => {
+          return this.ethereumClient
+            .getTransaction(hash)
+            .then((transaction) => {
+              if (!transaction) {
+                throw new Error('Transaction not found')
+              }
+              return transaction
+            })
+            .catch(retry)
+        },
+        { factor: 2, forever: false, retries: 10 }
+      )
+    } catch (error) {
+      console.error(error) // TODO: log properly
+      return null
+    }
+  }
+
+  private async waitForReceipt(hash: Hash256) {
+    return promiseRetry(
+      (retry) =>
+        this.ethereumClient.waitForTransactionReceipt(hash).catch(retry),
+      { forever: true }
```

Comment:
This also shouldn't be forever

Reply:
`waitForTransactionReceipt` from `ethers` waits for a transaction until it is mined and it should not throw unless something unexpected happens such as network error. My reasoning was that we'd rather wait longer than consider a mined/reverted transaction forgotten, because of something unexpected. Taking that into consideration I went with `{ forever: true }`, because I was not sure how long we should wait to safely consider a transaction forgotten in such case.